### DISCUSSION TITLE
Are there any gtk4 plugins in the planning?

### DISCUSSION THREAD

**Discussion by aprsa** [python-pillow/Pillow#6049] [Q&A] [answered]
Hi all, I'm using pillow to manipulate RGBA images; I'd like to incorporate them into the gtk4 front-end and I can't find an elegant way to transfer data between PIL.Image and gtk.Image/gtk.Picture. I see that pillow already has QT and TK plugins but none for GTK. Are there any plans (or any known workarounds) to make the two exchange data?

**Comment by radarhere** [comment, marked as answer]
There is no active plan for such functionality, no.

Searching, I found https://gitlab.gnome.org/GNOME/pygobject/-/issues/225
```python
from PIL import Image
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk, GdkPixbuf, GLib
topgtkwindow = Gtk.Window( title = "TEST" )
topgtkwindow.connect( "destroy", Gtk.main_quit )
gtkimage = Gtk.Image()
topgtkwindow.add( gtkimage )
topgtkwindow.show_all()
pillowimage = Image.new( "RGB", (800,600) )
glibbytes = GLib.Bytes.new( pillowimage.tobytes() )
gdkpixbuf = GdkPixbuf.Pixbuf.new_from_data( glibbytes.get_data(), GdkPixbuf.Colorspace.RGB, False, 8, pillowimage.width, pillowimage.height, len( pillowimage.getbands() )*pillowimage.width, None, None )
gtkimage.set_from_pixbuf( gdkpixbuf )
```

**Reply by aprsa** [reply]
Thanks for the quick response @radarhere! Yep, I found the same link, but gtk4 is moving away from pixbuf and I'm looking at Gdk.Texture.new_from_bytes() as a possible replacement. Should that be of any help to anyone, I'd be happy to post if/once I figure out how to make the two exchange the data nicely.
